### PR TITLE
[master] Add win_appx module and state

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds a new `win_appx` execution module and `win_appx` state module for managing Microsoft Store apps on Windows.

Adds the `dism.add_provisioned_package` execution function for adding provisioned packages to an image.
Adds the `dism.provisioned_packages` execution function for listing provisioned packages on an image.
Adds the `dism.provisioned_package_installed` state function for provisioning packages.

Moves the `_pshell_json` function in the `win_wusa` and `win_servermanager` modules to `salt.utils.win_pwsh.run_dict`. This function runs a powershell command and returns a dictionary of the output.

Adds and updates tests for these modules.

Updates the `tests/unit/modules/test_win_wusa.py` to pytest (`tests/pytests/unit/modules/tets_win_wusa.py`).

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/63978

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/win_appx.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+reinstalled when the system is updated.
+
+Apps removed with this module will remove the app for all users and deprovision
+the app. Deprovisioned apps will neither be installed for new users nor will
+they be upgraded.
+
+An app removed with this module can only be re-provisioned on the machine, but
+it can't be re-installed for all users. Also, once a package has been
+deprovisioned, the only way to reinstall it is to download the package. This is
+difficult. I've outlined the steps below:
```
Comment (s0undt3ch) [inline salt/modules/win_appx.py:16]:
```suggestion
difficult. The steps are outlined below:
```

File: salt/modules/win_appx.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    - Open the page for the app in the Microsoft Store
+    - Click the share button and copy the URL
+
+2. Look up the packages on https://store.rg-adguard.net/:
+    - Ensure ``URL (link)`` is selected in the first dropdown
+    - Paste the URL in the search field
+    - Ensure Retail is selected in the 2nd dropdown
+    - Click the checkmark button
+
+This should give you a list of URLs for the package and all dependencies for all
```
Comment (s0undt3ch) [inline salt/modules/win_appx.py:28]:
```suggestion
This should give a list of URLs for the package and all dependencies for all
```

**Review by s0undt3ch — COMMENTED** [review]
Better this way?

File: salt/modules/win_appx.py (hunk 1 above)
Comment (twangboy) [reply salt/modules/win_appx.py:16]: Yes... better

File: salt/modules/win_appx.py (hunk 2 above)
Comment (twangboy) [reply salt/modules/win_appx.py:28]: I changed it to `This should return a list...`

**Review by s0undt3ch — DISMISSED** [review]

**Review by whytewolf — DISMISSED** [review]

File: salt/modules/win_appx.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
+"""
+Manage provisioned apps
+=======================
+
+Provisioned apps are part of the image and are installed for every user the
```
Comment (Ch3LL) [inline salt/modules/win_appx.py:5]: Can you add a versionadded for 3007 for this new module

File: salt/modules/win_dism.py [hunk 4]
```diff
@@ -511,6 +514,58 @@ def add_package(
     return __salt__["cmd.run_all"](cmd)
 
 
+def add_provisioned_package(package, image=None, restart=False):
+    """
+    Provision a package using DISM. A provisioned package will install for new
```
Comment (Ch3LL) [inline salt/modules/win_dism.py:519]: Can you add a versionadded for 3007.0 for this new function?

File: salt/modules/win_dism.py [hunk 5]
```diff
@@ -654,6 +709,32 @@ def installed_packages(image=None):
     )
 
 
+def provisioned_packages(image=None):
+    """
+    List the packages installed on the system
```
Comment (Ch3LL) [inline salt/modules/win_dism.py:714]: same here, versionadded

File: salt/states/win_appx.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
+"""
+Manage Microsoft Store apps on Windows. Removing an app with this modules will
```
Comment (Ch3LL) [inline salt/states/win_appx.py:2]: same here versionadded for this new state module

File: salt/states/win_dism.py [hunk 7]
```diff
@@ -342,20 +387,102 @@ def package_installed(
     return ret
 
 
+def provisioned_package_installed(name, image=None, restart=False):
+    """
+    Install a package.
+
+    Args:
```
Comment (Ch3LL) [inline salt/states/win_dism.py:394]: here too

File: salt/utils/win_pwsh.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    Raises:
+        CommandExecutionError:
+            If an error is encountered or the command does not complete
+            successfully
+    """
+    if "convertto-json" not in cmd.lower():
+        cmd = f"{cmd} | ConvertTo-Json"
+    if "progresspreference" not in cmd.lower():
+        cmd = f"$ProgressPreference = 'SilentlyContinue'; {cmd}"
+    log.debug("PowerShell: %s", cmd)
```
Comment (Ch3LL) [inline salt/utils/win_pwsh.py:46]: Could this possibly log sensitive information?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/utils/win_pwsh.py (hunk 8 above)
Comment (twangboy) [reply salt/utils/win_pwsh.py]: I removed the log message. I believe ``cmd.run_all`` still logs the command being run though, so I'm not sure this fixes the issue.

File: salt/states/win_dism.py (hunk 7 above)
Comment (twangboy) [reply salt/states/win_dism.py:396]: Done

File: salt/states/win_appx.py (hunk 6 above)
Comment (twangboy) [reply salt/states/win_appx.py:2]: Done

File: salt/modules/win_dism.py (hunk 5 above)
Comment (twangboy) [reply salt/modules/win_dism.py:716]: Done

File: salt/modules/win_dism.py (hunk 4 above)
Comment (twangboy) [reply salt/modules/win_dism.py:519]: Done

File: salt/modules/win_appx.py (hunk 3 above)
Comment (twangboy) [reply salt/modules/win_appx.py:7]: Done

File: salt/utils/win_pwsh.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+def __virtual__():
+    """
+    Only load if windows
+    """
+    if not salt.utils.platform.is_windows():
+        return False, "This utility will only run on Windows"
+
+    return __virtualname__
+
```
Comment (s0undt3ch) [inline salt/utils/win_pwsh.py:17]:
```suggestion
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/win_pwsh.py (hunk 9 above)
Comment (s0undt3ch) [reply salt/utils/win_pwsh.py:17]: Do not make this change. It has other implications.

**Review by s0undt3ch — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**